.github/codecov.yaml

```
@@ -0,0 +1,59 @@
+# Find more at https://docs.codecov.com/docs/codecovyml-reference
+codecov:
+  bot: "Codecov Bot"
+  max_report_age: 12
+  require_ci_to_pass: yes
+  notify:
+    after_n_builds: 1
+    wait_for_ci: yes
+
+# Layout of the PR comment produced by Codecov bot
+comment:
+  layout: "header, diff, flags, components, files"
+
+ignore:
+  - api/external/** #  ignoring external vendor code
```

```suggestion
  - api/external/** #  ignoring external vendor code
  - **/*.deepcopy.go # ignore controller-gen generated code 
```

nit: we should probably ignore controller-gen generated code as otherwise the unit test coverage will probably be skewed by them 🤔